src/cljs/chronograph_web/db.cljs

```diff
@@ -3,3 +3,53 @@
 (def default-db {:user {;; signin-state can be:
                         ;; :signed-in, :signed-out or :fetching-profile
                         :signin-state :fetching-profile}})
+
+(def conj-to-set (fnil conj #{}))
+
+(defn add-to-set-in
+  "Adds v to a set in the db at the given path."
+  [db path v]
+  (update-in db
+             path
+             conj-to-set
+             v))
+
+(defn current-page-name
+  [db]
+  (get-in db [:page :handler]))
+
+;; Page state functions
+;; Use these functions to get and set state which is specific to
+;; the current page.
+
+(defn get-in-page-state
+  [db path]
+  (get-in db (concat [:page-state (current-page-name db)] path)))
```

Comment:
I try to use `into` instead of `concat` as a matter of general practice, because the latter is lazy and can cause those pesky problems with unrealised lazy seqs. 

Thoughts?

Reply:
Don't really think it matters in this case, since the sequence is small and it's being thrown away almost immediately.

Reply:
Like I noted, I am curious about general practice, not this specific instance.

For example, in this particular ns, there are at least two places where a `concat`'d collection is being added into the DB. 

I'm not aware if reagent, or some other mechanism in re-frame's event loop, force-realizes all the things being put in the ratom. If yes, then well and good. If not, could there arise scenarios where a lazy seq ends up in the db, and something tries to make an HTTP request, or write to local storage or some other side-effecty thing that fails on lazy seqs?

Reply:
Having a lazy sequence in the DB should not pose a problem in any case because "using" it will realize the lazy seq. They are a pretty good abstraction in that regard. 

The only time one has to be careful with lazy seqs is when the lazy seq is performing side effects (for example, a `map` which is querying postgres for each item in a sequence). Which is not the case here, and should definitely not be the case for anything we put in the re-frame DB. If a lazy seq is being produced through pure means, then it doesn't matter when it's realized. If it's being produced through impure means, then one has to worry about when and where it is realized.

Reply:
Also in this namespace, `concat` is only being used to construct paths. The `concat`ed collection is not itself put into the DB.

Reply:
> Also in this namespace, concat is only being used to construct paths. The concated collection is not itself put into the DB.

Ugh. This means I misread the code twice. Sorry about that.

Also:

> If it's being produced through impure means, then one has to worry about when and where it is realized.

Right, that's the important distinction, and I was thinking about it poorly.

Thanks!